qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/ScaleResiliencyTest.java

```diff
@@ -0,0 +1,230 @@
+        TestCluster.builder()
+            .useRecordingExporter(true)
+            .withGatewaysCount(1)
+            .withGatewayConfig(
+                g ->
+                    g.gatewayConfig()
+                        .getCluster()
+                        .getMembership()
+                        // Decrease the timeouts for fast convergence of gateway topology.
+                        .setSyncInterval(Duration.ofSeconds(1))
+                        .setFailureTimeout(Duration.ofSeconds(2)))
+            .withEmbeddedGateway(false)
+            .withBrokersCount(initialClusterSize)
+            .withPartitionsCount(partitionsCount)
+            .withReplicationFactor(replicationFactor)
+            .withBrokerConfig(
+                b -> {
+                  b.brokerConfig()
+                      .getExperimental()
+                      .getFeatures()
+                      .setEnableDynamicClusterTopology(true);
+                  b.brokerConfig()
+                      .getCluster()
+                      .getMembership()
+                      // Decrease the timeouts for fast convergence of gateway topology.
+                      .setSyncInterval(Duration.ofSeconds(1))
+                      .setFailureTimeout(Duration.ofSeconds(2));
+                  b.withWorkingDirectory(
+                      getDataDirectory(tmpDir, b.brokerConfig().getCluster().getNodeId()));
+                })
+            .build();
+    cluster.start();
+    cluster.awaitCompleteTopology();
+    return cluster;
+  }
+
+  @Nested
+  class ScaleUp {
+    private static final int REPLICATION_FACTOR = 2;
+    private static final int PARTITIONS_COUNT = 2;
+    private static final int INITIAL_CLUSTER_SIZE = 2;
+    @TempDir Path tmpDir;
+    private final List<TestStandaloneBroker> newBrokers = new ArrayList<>();
+    private TestCluster cluster;
+
+    @BeforeEach
+    void setup() {
+      cluster = createCluster(INITIAL_CLUSTER_SIZE, PARTITIONS_COUNT, REPLICATION_FACTOR, tmpDir);
+    }
+
+    @AfterEach
+    void shutdownNewBrokers() {
+      newBrokers.forEach(TestStandaloneBroker::close);
+      newBrokers.clear();
+      cluster.shutdown();
+    }
+
+    @Test
+    void shouldContinueScaleAfterRestart() {
+      // given cluster size 2, partitions 2, replicationFactor 2;
+
+      // shutdown broker 1
+      cluster.brokers().get(MemberId.from("1")).stop();
+      final var newClusterSize = INITIAL_CLUSTER_SIZE + 1;
+      final var newBroker =
+          createNewBroker(
+              newClusterSize, newClusterSize - 1, getDataDirectory(tmpDir, newClusterSize - 1));
+
+      // scale to 3
+      Utils.scale(cluster, newClusterSize);
+      // wait until broker 2 is added to the cluster
+      Awaitility.await()
+          .untilAsserted(
+              () -> ClusterActuatorAssert.assertThat(cluster).hasActiveBroker(newClusterSize - 1));
+      // applying operations is stuck because there is no leader for partitions, so partitions
```

Comment:
This depends on the partitions distribution, right? because in general there _are_ two brokers, it's just that the newly started one isn't member of either of them.

Reply:
> This depends on the partitions distribution, right?
Generally yes. But here replication factor is 2. Since broker 1 is stopped now, it is guaranteed that there is no leader for both partitions because broker 2 has not joined yet.